app/controllers/hyku/api/v1/sessions_controller.rb

```diff
@@ -69,11 +69,15 @@ def default_cookie_options
             {
               path: '/',
               same_site: :lax,
-              domain: ('.' + request.host),
+              domain: ('.' + cookie_domain),
               secure: Rails.env.production?,
               httponly: true
             }
           end
+
+          def cookie_domain
+            @account.frontend_url.presence || request.host
```

Comment:
I think this needs to use safe navigation in case the Account doesn't have `frontend_url` available.
```suggestion
            @account&.frontend_url.presence || request.host
```

Reply:
Not sure about that. The `@account` should always be present after a login and we set the cookies after a successful login or on token refresh. Both should have it present, right?

Reply:
`frontend_url` hasn't been declared on the Account AFAICT.  Is there supposed to be a DB migration as part of this PR?
```
Failure/Error: @account.frontend_url.presence || request.host

NoMethodError:
  undefined method `frontend_url' for #<Account:0x000055d381fd5548>
```

Reply:
Oh, right, so safe navigation doesn't work here...`try` is probably better:
```suggestion
            @account.try(:frontend_url).presence || request.host
```

Reply:
It's part of https://github.com/ubiquitypress/hyku_addons/pull/92/files

Reply:
Ah, I see.  I had forgotten about that PR.  The problem here is that `hyku-api` doesn't know anything about `hyku_addons` and the internal test application isn't the same as `hyku_addons` in that it doesn't include `hyku_addons` at all.  The shortest path is probably to use `try` so it will work when available, but you'll have to mock it in your tests here because it isn't defined.

Reply:
I am not a big fan of `:try` and try to avoid it if possible and it wouldn't help because if there is no `frontend_url` column it would raise a NoMethodError, which :try doesn't handle, right?.

I've used `current_account.attributes.with_indifferent_access[:frontend_url].presence || request.host` which is longer but safer. WDYT?

Reply:
You are right! I didn't know `try` handles NoMethodError not just if the receiver is nil, but even if it doesn't answer to it!.
The worse part is that it's been like that since Rails 4...